File: java-pi-pantry/src/test/java/com/langrsoft/pi/pantry/AnItemRetriever.java
package com.langrsoft.pi.pantry;

import com.langrsoft.util.HttpClient;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import static com.langrsoft.util.JsonUtil.toJson;
import static org.hamcrest.Matchers.equalTo;
import static org.hamcrest.Matchers.is;
import static org.junit.Assert.assertThat;
import static org.mockito.Matchers.matches;
import static org.mockito.Mockito.when;

@RunWith(MockitoJUnitRunner.class)
public class AnItemRetriever {
    @InjectMocks
    ItemRetriever itemRetriever;
    @Mock
    HttpClient httpClient;

    @Test
    public void parsesResponseJsonToItem() {
        String wheatiesUpc = "0016000275652";
        String responseText = toJson(new ItemBuilder("Wheaties").create());
        when(httpClient.retrieveText(matches("http://.*/json/.*/" + wheatiesUpc)))
                .thenReturn(responseText);

        Item item = itemRetriever.retrieve(wheatiesUpc);

        assertThat(item.getName(), is(equalTo(("Wheaties"))));
    }
}

File: csharp-pi-pantry/csharp-pi-pantry-test/DomainTests/AnItemRetriever.cs
using Pipantry.Domain;
using Pipantry.Util;
using NUnit.Framework;
using Moq;

namespace Test.Pipantry.Domain
{
    [TestFixture]
    public class AnItemRetriever
    {
        ItemRetriever itemRetriever;
        Mock<Http> httpClient;

        [SetUp]
        public void create()
        {
            httpClient = new Mock<Http>();
            itemRetriever = new ItemRetriever(httpClient.Object);
        }

        [Test]
        public void ParsesResponseJsonToItem()
        {
            var wheatiesUpc = "0016000275652";
            var responseText = JsonUtil.ToJson(new Item("Wheaties"));

            httpClient.Setup(client => client.RetrieveText(It.IsRegex("http://.*/json/.*/" + wheatiesUpc)))
                .Returns(responseText);

            var item = itemRetriever.Retrieve(wheatiesUpc);

            Assert.That(item.Name, Is.EqualTo(("Wheaties")));
        }
    }
}
